test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -258,19 +257,26 @@ bool ThreeChannelRequestUsingSameRemoteId()
     return ((channelOperationResult2 == WPN_E_OUTSTANDING_CHANNEL_REQUEST) && (channelOperationResult3 == WPN_E_OUTSTANDING_CHANNEL_REQUEST));
 }
 
-bool RegisterActivator()
+bool RegisterActivator(winrt::Microsoft::Windows::PushNotifications::PushNotificationRegistrationToken token)
 {
+    PushNotificationManager::UnregisterActivator(token, PushNotificationRegistrationKind::PushTrigger | PushNotificationRegistrationKind::ComActivator);
     try
     {
         PushNotificationActivationInfo info(
             PushNotificationRegistrationKind::PushTrigger | PushNotificationRegistrationKind::ComActivator,
-            winrt::guid("00000000-0000-0000-0000-000000000001"));
+            c_fakeComServerId);
 
         auto token = PushNotificationManager::RegisterActivator(info);
-        if (token.Cookie() == 0 || token.TaskRegistration() == nullptr)
+        if (token.Cookie() == 0)
+        {
+            return false;
```

Comment:
Why would this return 0?

Reply:
I thought that CoRegisterClassObject in PushNotificationManager would return 0 in the case of failing to register but we already have a THROW_IF_FAILED wrapping that function, so it looks like we don't to check for Cookie being 0 here.